### PR TITLE
feat(storage): storage metrics with table_id label

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Reorganize the original metrics to separate different metrics and avoid mixing them. Add table_id level label for some performance metrics

**This section will be used as the commit message. Please do not leave this empty!**

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- separate different metrics for `storage`, `state_store`, `compactor`
- table_id level label 

## Checklist

~- [ ] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/storage/src/hummock/store/version.rs
Comment (hzxa21) [inline src/storage/src/hummock/store/version.rs:474]: Shall we let `report` take a table id str to avoid multiple to_string for the same table_id?

File: src/storage/src/monitor/mod.rs
Comment (hzxa21) [inline src/storage/src/monitor/mod.rs:17]: remove

File: src/storage/src/monitor/mod.rs
Comment (hzxa21) [inline src/storage/src/monitor/mod.rs:38]: remove

File: src/storage/src/store_impl.rs
Comment (hzxa21) [inline src/storage/src/store_impl.rs:111]: in_memory shouldn't need hummock specific `StateStoreMetrics`.

File: src/storage/src/store_impl.rs
Comment (hzxa21) [inline src/storage/src/store_impl.rs:146]: sled shouldn't need hummock specific `StateStoreMetrics`.

File: src/storage/src/store_impl.rs
Comment (hzxa21) [inline src/storage/src/store_impl.rs:155]: shared_in_memory_store shouldn't need hummock specific `StateStoreMetrics`.

**Review by hzxa21 — COMMENTED** [review]
We should also invoke `remove_label_values` when all state stores related to a table_id are destroyed. We can leave it to a future PR.

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: src/storage/src/monitor/monitored_store.rs
Comment (Li0k) [reply src/storage/src/monitor/monitored_store.rs:42]: yes, MonitoredStateStore not need the hummock specific `StateStoreMetrics`. But we get the local_state_store by `new_local` function through MonitoredStateStore, so the `StateStoreMetrics` is just a parameter maintain by MonitoredStateStore (to maintain registry)

**Comment by Li0k** [comment]
@hzxa21 fix comments, PTAL

File: src/storage/src/monitor/state_store_metrics.rs
Comment (hzxa21) [inline src/storage/src/monitor/state_store_metrics.rs:29]: Can we change the file name as well?

File: src/storage/src/monitor/local_metrics.rs
Comment (hzxa21) [inline src/storage/src/monitor/local_metrics.rs:77]: table_id_label can just ba `&str` and we can use a empty string for use cases without a table id

**Review by hzxa21 — APPROVED** [review]
Rest LGTM.

**Comment by hzxa21** [comment]
cc @huangjw806 This PR modifies the name of several storage metrics as well as the grafana dashboard. We may need to update the grafana dashboard for the testing pipeline accordingly after merging this PR.

**Comment by huangjw806** [comment]
> cc @huangjw806 This PR modifies the name of several storage metrics as well as the grafana dashboard. We may need to update the grafana dashboard for the testing pipeline accordingly after merging this PR.

Sure，I will update the cloud grafana dashboard after this pr merged.

**Pull request merged**
